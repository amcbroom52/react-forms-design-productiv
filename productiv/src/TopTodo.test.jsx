import { describe, it, expect } from "vitest";

import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";
import Todo from "./Todo";

const TODOS = [
    {
    id: '1',
    title: "title1",
    description: "description1",
    priority: 1
    },
    {
    id: '2',
    title: "title2",
    description: "description2",
    priority: 2
    },
    {
    id: '3',
    title: "title3",
    description: "description3",
    priority: 3
    }
];



describe("Todo component", function () {
    it("renders without crashing", function () {
        render(<TopTodo todos={TODOS} />);
    });

    it("matches snapshot", function () {
        const { asFragment } = render(<TopTodo todos={TODOS} />);
        expect(asFragment()).toMatchSnapshot();
    })

    it("contains correct TopTodo values", function() {
        const {container} = render(<TopTodo todos={TODOS} />);
        const topTodo = render(<Todo todo={TODOS[0]} />);
        expect(container).toEqual(topTodo.container);
        expect(container).not.toContainHTML("description2");
    });
})