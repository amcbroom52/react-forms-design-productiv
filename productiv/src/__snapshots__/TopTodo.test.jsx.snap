// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Todo component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
        title1
      </b>
       
      <small>
        Priority: 1
      </small>
    </div>
    <div>
      <small>
        description1
      </small>
    </div>
  </div>
</DocumentFragment>
`;
